packages/compiler/formatter/print/printer.ts

```diff
@@ -1002,22 +999,23 @@ export function printModelProperty(
   ];
 }
 
-function isStringSafeToUnquote(id: StringLiteralNode, options: CadlPrettierOptions): boolean {
-  const unquotedRawText = getRawText(id, options).slice(1, -1);
-  if (id.value !== unquotedRawText) {
-    return false;
+function printIdentifier(id: IdentifierNode) {
+  return needBacktick(id) ? `\`${id.sv}\`` : id.sv;
+}
+
+function needBacktick(id: IdentifierNode) {
```

Comment:
I realized there is one issue with completion (IntelliSense) also needing to check if backticks need to be inserted. Right now, it will happily complete `Foo.a-b-c` even if it needs to be Foo.`a-b-c` to compile.

Reply:
good catch, will fix

Reply:
fixed completetion